src/includes/class_reservationPermissions.php

```diff
@@ -1,76 +1,62 @@
 <?php
 class reservationPermissions {
 
-  public function getRecords($id = null){
-      try {
-          // call engine
-          $engine    = EngineAPI::singleton();
-          $localvars = localvars::getInstance();
-          $db        = db::get($localvars->get('dbConnectionName'));
-          $sql       = "SELECT * FROM `reservePermissions`";
-          $validate  = new validate;
+  private $localvars;
+  private $engine;
+  private $this->db;
+  private $validate;
+
+  function __construct() {
+    $this->engine    = EngineAPI::singleton();
+    $this->localvars = localvars::getInstance();
+    $this->db        = db::get($this->localvars->get('dbConnectionName'));
+    $this->validate  = new validate;
+  }
 
-          if (isset($id)) {
-           $id        = dbSanitize($id);
-          }
+  public function getRecords($id = null){
+    $sql = "SELECT * FROM `reservePermissions`";
+    try {
+      if(!isnull($id) && !$this->validate->integer($id)){
+        throw new Exception("Invalid ID provided.");
+      }
 
```

Comment:
We don't need to sanitize the variable here, we don't even know if we are going to use it yet. 

First, lets see if we have it (we should always check that we have it before we validate it!!) ... and if we have it, we validate it.